src/integration/resources/inprocess/cluster.go

```diff
@@ -233,6 +238,66 @@ func NewClusterFromSpecification(
 	return m3, nil
 }
 
+// ConfigurePlacementsForAggregation sets up the correct placement information for
+// coordinators and aggregators when aggregation is enabled.
+func ConfigurePlacementsForAggregation(
+	nodes resources.Nodes,
+	coord resources.Coordinator,
+	aggs resources.Aggregators,
+	specs ClusterSpecification,
+	opts resources.ClusterOptions,
+) error {
+	if len(aggs) == 0 {
+		return nil
+	}
+
+	coordAPI := coord
+	hostDetails, err := coord.HostDetails()
+	if err != nil {
+		return err
+	}
+
+	// With remote aggregation enabled, aggregation is not handled within the coordinator.
+	// When this is true, the coordinator will fail to start until placement is updated with
+	// aggregation related information. As such, use the coordinator embedded within the dbnode
+	// to configure the placement and topics.
+	if specs.Configs.Coordinator.Downsample.RemoteAggregator != nil {
+		if len(specs.Configs.DBNodes) == 0 ||
+			specs.Configs.DBNodes[0].Coordinator == nil {
+			return errors.New("remote aggregation requires at least one DB node" +
+				" running an embedded coordinator for placement and topic configuration")
+		}
+
+		embedded, err := NewEmbeddedCoordinator(nodes[0].(*DBNode))
+		if err != nil {
+			return nil
+		}
+
+		coordAPI = embedded
```

Comment:
If the embedded coordinator is used here, are we assuming the coordinator in the cluster specification is always using m3db for the kv store?

Reply:
Not exactly. If the coordinator in the cluster specification wants to use a different KV store, then they'll just need to update the kv store used in the embedded coordinators config. Practically though, I don't see anyone changing to configs to modify which kvstore to use (even though we support it)